### PR TITLE
Added some expanded cases

### PR DESCRIPTION
**Pull request by 404Naz**
Added cases for adding Expressions and exponent division.

### CODE REVIEW

**Comment by AnthonyN1** [comment]
We love comparing signed and unsigned types.

**Comment by 404Naz** [comment]
I did not change anything in divide but the tests have broken...not sure if something I added or changed did something or they were always broken.

File: src/Add.cpp [hunk 1]
```diff
@@ -85,30 +84,157 @@ auto Add<Expression>::Simplify() const -> std::unique_ptr<Expression>
         }
     }
 
-    std::map<std::unique_ptr<Expression>, unsigned> terms;
+    if (auto exprCase = Add<Expression>::Specialize(simplifiedAdd); exprCase != nullptr) {
+        if (exprCase->GetMostSigOp().Equals(exprCase->GetLeastSigOp())) {
+            return std::make_unique<Multiply<Expression>>(Real { 2.0 }, exprCase->GetMostSigOp());
+        }
+    }
```
Comment (matthew-mccall) [inline src/Add.cpp:91]: This is good. Try using more descriptive variable names such as `twoSameExprCase` instead. Also, note that you're working with the `Generalized version of `Add` so there's no need to `Specialize` it into `Add<Expression>`

**Review by matthew-mccall — CHANGES_REQUESTED** [review]

File: src/Add.cpp (hunk 1 above)
Comment (matthew-mccall) [reply src/Add.cpp:91]: While the above is one example, you specialize to `Expression` children quite often. Please keep in mind that is not necessary as we are already working in the `Generalize`d version of this expression. Also there are some redundant simplifications with adding `Exponent` and `Imaginary`. Namely the cases: $i+i=2i$ and $x^y+x^y=2x^y$

**Comment by matthew-mccall** [comment]
Closed due to inactivity

**Pull request closed without merging**